### PR TITLE
Allow to change app. name using CMake cache (#34,  #71)

### PR DESCRIPTION
**Pull request by ghost**
- [x] Allow to change `AppName` in `config.h`
- [x] Replace most hardcoded strings in other files
- [x] Replace Qt application name
- [ ] ~~Replace name in translations~~
- [x] Rebase on `dev` to allow automatic merge

### CODE REVIEW

**Comment by crackedmind** [comment]
:white_check_mark: [Build kepka 1.0.56-xivhvble completed](https://ci.appveyor.com/project/procxx/tdesktop/build/1.0.56-xivhvble) (commit https://github.com/procxx/kepka/commit/10f1df9b19 by @foxcpp)

**Comment by ghost** [comment]
>- [ ] Replace name in translations

Requires more work than I expected. I think it's ok to merge without this.

File: Telegram/CMakeLists.txt [hunk 1]
```diff
@@ -164,6 +164,18 @@ foreach (src ${style_files})
 
 endforeach()
 
+##================================================
+## Branding
+##================================================
+
+set(TDESKTOP_APPNAME_OLD "Kepka Win (Unofficial)"
```
Comment (berkus) [inline Telegram/CMakeLists.txt]: Can we please drop the _OLD things?

File: Telegram/CMakeLists.txt [hunk 2]
```diff
@@ -164,6 +164,18 @@ foreach (src ${style_files})
...
+    CACHE STRING "(Deprecated) Used as old profile name from earlier versions")
+set(TDESKTOP_APPNAME "Kepka"
```
Comment (berkus) [inline Telegram/CMakeLists.txt]: TDESKTOP_ is probably a bad variable prefix for KEPKA_ variables.

Rename it?

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp [hunk 3]
```diff
@@ -456,13 +456,14 @@ void psRegisterCustomScheme() {
 			s.setCodec("UTF-8");
 			s << "[Desktop Entry]\n";
 			s << "Version=1.0\n";
-			s << "Name=Telegram Desktop\n";
+			s << "Name=" << str_const_toString(AppName) << '\n';
 			s << "Comment=Official desktop version of Telegram messaging app\n";
```
Comment (berkus) [inline Telegram/SourceFiles/platform/linux/specific_linux.cpp]: Update comment to Unofficial version

**Review by berkus — CHANGES_REQUESTED** [review]

**Comment by crackedmind** [comment]
:white_check_mark: [Build kepka 1.0.60-jcqktpec completed](https://ci.appveyor.com/project/procxx/tdesktop/build/1.0.60-jcqktpec) (commit https://github.com/procxx/kepka/commit/2078b912ad by @foxcpp)

**Comment by ghost** [comment]
Ping?

**Review by berkus — APPROVED** [review]
Some more improvements could be done, but lets not freeze it to death here.

**Pull request merged**